lib/clockwise_import.rb

```diff
@@ -0,0 +1,211 @@
+require 'csv'
+
+class ClockwiseImporter
+  def initialize(csv_path:, tenant:)
+    @csv_path = csv_path
+    @tenant = tenant
+    Apartment::Tenant.switch @tenant
+  end
+
+  def import!
+    Entry.transaction do
+      CSV.foreach(@csv_path,
+                  headers: true,
+                  quote_char: '"',
+                  col_sep: ';') do |r|
+        row = ImportRow.new(r)
+        EntryBuilder.new(row).create! unless row.hours < 1
+      end
+    end
+  end
+
+  def cleanup!
+    ActiveRecord::Base.transaction do
+      Project.where("name in (?)", PROJECTS_TO_REMOVE).each do |project|
+        project.entries.each do |entry|
+          entry.destroy!
+        end
+        project.destroy!
+      end
+      Category.find_each do |cat|
+        if cat.entries.length == 0
+          cat.destroy!
+        end
+      end
+      Tag.find_each do |tag|
+        if tag.entries.length == 0
+          tag.destroy!
+        end
+      end
+      Client.find_each do |client|
+        if client.projects.length == 0
+          client.destroy!
+        end
+      end
+    end
+  end
+
+  def reset_updated_at!
+    ActiveRecord::Base.transaction do
+      Project.find_each do |project|
+        project.update_attributes(
+          updated_at: find_most_recent_entry_date(project)) unless project.entries.length == 0
+      end
+    end
+  end
+
+  def find_most_recent_entry_date(project)
+    date = project.entries.order(:date).last.date
+    DateTime.parse date.to_s
+  end
+
+end
+
+class EntryBuilder
+  def initialize(row)
+    @row = row
+  end
+
+  def create!
+    client = build_client
+    project = build_project(client)
+    category = build_category
+    user = build_user
+    Entry.create!(
+      project: project,
+      category: category,
+      user: user,
+      hours: @row.hours,
+      date: @row.date,
+      description: @row.description,
+      billed: @row.archive
+    )
+  end
+
+  private
+
+  def build_client
+    Client.where('lower(name) = ?', @row.client.downcase).
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.